### PR TITLE
Some adjustments before RC2

### PR DESCRIPTION
**Pull request by NicolasEYSSERIC**
- Fixing some NullPointerException
- Removing useless code on tabs
- Integrating optimizations already done on Silverpeas 5 (except online edition)

### CODE REVIEW

File: core-services/questioncontainer/src/main/java/org/silverpeas/core/questioncontainer/container/service/DefaultQuestionContainerService.java [hunk 1]
```diff
@@ -29,6 +29,7 @@
 import org.silverpeas.core.admin.service.OrganizationController;
 import org.silverpeas.core.admin.service.OrganizationControllerProvider;
 import org.silverpeas.core.admin.space.SpaceInst;
```
Comment (silveruser) [inline core-services/questioncontainer/src/main/java/org/silverpeas/core/questioncontainer/container/service/DefaultQuestionContainerService.java]: MINOR Remove this unused import 'org.silverpeas.core.admin.space.SpaceInst'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-library/src/main/java/org/silverpeas/core/contribution/content/form/field/AccessPathField.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
 
 import org.silverpeas.core.admin.service.OrganizationControllerProvider;
 import org.silverpeas.core.admin.space.SpaceInst;
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/contribution/content/form/field/AccessPathField.java]: MINOR Remove this unused import 'org.silverpeas.core.admin.space.SpaceInst'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-library/src/main/java/org/silverpeas/core/contribution/content/form/field/ExplorerField.java [hunk 3]
```diff
@@ -26,6 +26,7 @@
 import org.silverpeas.core.ForeignPK;
 import org.silverpeas.core.admin.service.OrganizationControllerProvider;
 import org.silverpeas.core.admin.space.SpaceInst;
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/contribution/content/form/field/ExplorerField.java]: MINOR Remove this unused import 'org.silverpeas.core.admin.space.SpaceInst'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-library/src/main/java/org/silverpeas/core/admin/service/Admin.java [hunk 4]
```diff
@@ -2735,44 +2732,57 @@ public String identify(String sKey, String sSessionId, boolean isAppInMaintenanc
       List<String> result = new ArrayList<>();
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/admin/service/Admin.java]: MAJOR Remove this unused "result" local variable. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1481)

File: core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/browsebars/BrowseBarComplete.java [hunk 5]
```diff
@@ -28,6 +28,7 @@
 import org.silverpeas.core.admin.service.OrganizationController;
 import org.silverpeas.core.admin.service.OrganizationControllerProvider;
 import org.silverpeas.core.admin.space.SpaceInst;
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/browsebars/BrowseBarComplete.java]: MINOR Remove this unused import 'org.silverpeas.core.admin.space.SpaceInst'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/ComponentPathTag.java [hunk 6]
```diff
@@ -29,6 +29,7 @@
 import org.silverpeas.core.admin.service.OrganizationController;
 import org.silverpeas.core.admin.service.OrganizationControllerProvider;
 import org.silverpeas.core.admin.space.SpaceInst;
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/ComponentPathTag.java]: MINOR Remove this unused import 'org.silverpeas.core.admin.space.SpaceInst'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/window/WindowWithContextualDiv.java [hunk 7]
```diff
@@ -27,6 +27,7 @@
 import org.silverpeas.core.admin.service.OrganizationController;
 import org.silverpeas.core.admin.service.OrganizationControllerProvider;
 import org.silverpeas.core.admin.space.SpaceInst;
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/window/WindowWithContextualDiv.java]: MINOR Remove this unused import 'org.silverpeas.core.admin.space.SpaceInst'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-library/src/main/java/org/silverpeas/core/admin/service/DefaultOrganizationController.java [hunk 8]
```diff
@@ -35,6 +35,9 @@
 import org.silverpeas.core.admin.component.model.ComponentSearchCriteria;
 import org.silverpeas.core.admin.component.model.PersonalComponentInstance;
 import org.silverpeas.core.admin.component.model.SilverpeasComponentInstance;
+import org.silverpeas.core.admin.component.model.SilverpeasSharedComponentInstance;
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/admin/service/DefaultOrganizationController.java]: MINOR Remove this unused import 'org.silverpeas.core.admin.component.model.SilverpeasSharedComponentInstance'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-library/src/main/java/org/silverpeas/core/admin/service/DefaultOrganizationController.java [hunk 9]
```diff
@@ -35,6 +35,9 @@
...
+import org.silverpeas.core.admin.component.model.WAComponent;
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/admin/service/DefaultOrganizationController.java]: MINOR Remove this unused import 'org.silverpeas.core.admin.component.model.WAComponent'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-library/src/main/java/org/silverpeas/core/admin/service/DefaultOrganizationController.java [hunk 10]
```diff
@@ -35,6 +35,9 @@
...
+import org.silverpeas.core.admin.component.service.SilverpeasComponentInstanceProvider;
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/admin/service/DefaultOrganizationController.java]: MINOR Remove this unused import 'org.silverpeas.core.admin.component.service.SilverpeasComponentInstanceProvider'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by silveruser** [comment]
SonarQube analysis reported 13 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 10 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DefaultOrganizationController.java#L625](https://github.com/Silverpeas/Silverpeas-Core/blob/90b9e46ccaa45a7714725eae572a0b11ee13fbf9/core-library/src/main/java/org/silverpeas/core/admin/service/DefaultOrganizationController.java#L625): Remove this unused private "idAsInt" method. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [DefaultOrganizationController.java#L640](https://github.com/Silverpeas/Silverpeas-Core/blob/90b9e46ccaa45a7714725eae572a0b11ee13fbf9/core-library/src/main/java/org/silverpeas/core/admin/service/DefaultOrganizationController.java#L640): Remove this unused private "idAsString" method. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [MyLinksResource.java#L27](https://github.com/Silverpeas/Silverpeas-Core/blob/90b9e46ccaa45a7714725eae572a0b11ee13fbf9/core-web/src/main/java/org/silverpeas/core/webapi/mylinks/MyLinksResource.java#L27): Remove this unused import 'org.silverpeas.core.admin.space.SpaceInst'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SilverYoCha** [comment]
Merged.
Closed manually because of manual git operations performed during integration.

**Pull request closed without merging**